crates/bevy_asset/src/server/info.rs

```diff
@@ -61,7 +61,7 @@ impl AssetInfo {
 
 #[derive(Default)]
 pub(crate) struct AssetInfos {
-    path_to_id: HashMap<AssetPath<'static>, UntypedAssetId>,
+    path_to_id: HashMap<AssetPath<'static>, HashMap<TypeId, UntypedAssetId>>,
```

Comment:
Agreed, but I'll leave it off this PR, as the `TypeIdMap` type definition is private to the `bevy_ecs` crate. Ideally I think it would be moved into `bevy_utils`, since it has uses outside of the ECS crate.